frontend/ui/widget/touchmenu.lua

```diff
@@ -961,4 +975,227 @@ function TouchMenu:onBack()
+
+            local kv_pairs = {}
+            for i = 1, #found_menu_items do
+                table.insert(kv_pairs, { found_menu_items[i][1],
+                                         "", --found_menu_items[i][3],
+                                         callback = function() item_callback(i) end,
+                                         hold_callback = function() item_hold_callback(i) end,
+                })
+            end
+            return kv_pairs
+        end -- get_current_search_results()
+
+        if #found_menu_items > 0 then
+            self.kv = KeyValuePage:new{
+                title = _("Search results"),
+                kv_pairs = get_current_search_results(found_menu_items),
+            }
+            UIManager:show(self.kv)
+        else
+            UIManager:show(InfoMessage:new{
+                text = T(_("No menus containing '%1' found."), search_string),
+            })
+        end
+    end -- show_search_results()
+
+    local search_dialog
+    search_dialog = InputDialog:new{
+        title = _("Search menu entry"),
+        description = _("Search for a menu entry containing the following text (case insensitive)."),
+        input = G_reader_settings:readSetting("menu_search_string", _("Help")),
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    id = "close",
+                    callback = function()
+                        UIManager:close(search_dialog)
+                    end,
+                },
+                {
+                    text = _("Search"),
+                    callback = function()
+                        local search_for = search_dialog:getInputText()
+                        local status, err = pcall( function() ("test_string"):find(search_for) end)
+                        if status then
+                            search_for = Utf8Proc.lowercase(search_for)
+                            G_reader_settings:saveSetting("menu_search_string", search_for)
+                            UIManager:close(search_dialog)
+                            show_search_results(search_for)
+                        else
+                            err = err:sub(err:find("lua") + 10)  -- 10 = strlen("lua:1165: ")
+                            UIManager:show(InfoMessage:new{
+                                text = T(_("Malformed message:\n%1"), err)
+                            })
+                        end
+                    end,
+                },
+            }
+        },
+    }
+
+    -- Extensive_search will enable to scan sub_item_table_func() results too.
+    -- This can take (with a lot of system fonts enabled) quite some time and should not be done.
+    --- @todo Maybe caching of sub_item_table_func() results can improve things?
+    local extensive_search = G_reader_settings:readSetting("search_menu_extensive", false)
+    local check_button_restrict = CheckButton:new{
+        text = _("Extensive search"),
```

Comment:
Over-uxineering :/
The user using your new feature is not experienced: don't ask him to decide for an extensive or not search.
If it takes time, so let it take time: it's not a feature that needs to be reactive - and it give the user the feeling we're doing a lot of work to satisfy his request - and he won't have to do a 2nd query with this checked if he get no result with this unchecked.
If the font menu would take time, just ignore it - add some explicite code in here to detect and skip it.